From 56b27205d3e171f67aca434862f595c44590092c Mon Sep 17 00:00:00 2001
From: Bo Shen <voice.shen@atmel.com>
Date: Thu, 22 Jan 2015 17:25:54 +0800
Subject: [PATCH] driver: nandflash: make code more tight

As the nand flash id can share with at91sam9260ek,
so no need repeat them for at91sam9260ek.

Signed-off-by: Bo Shen <voice.shen@atmel.com>
---
 driver/nandflash.c | 15 +--------------
 1 file changed, 1 insertion(+), 14 deletions(-)

diff --git a/driver/nandflash.c b/driver/nandflash.c
index 6222079..c14fd20 100644
--- a/driver/nandflash.c
+++ b/driver/nandflash.c
@@ -48,7 +48,6 @@ static struct nand_chip nand_ids[] = {
 };
 #else
 static struct nand_chip nand_ids[] = {
-#ifdef CONFIG_AT91SAM9260EK
 	/* Samsung K9F2G08U0M 256MB */
 	{0xecda, 0x800, 0x20000, 0x800, 0x40, 0x0},
 	/* Samsung K9F2G08U0A 256MB */
@@ -61,23 +60,11 @@ static struct nand_chip nand_ids[] = {
 	{0x2caa, 0x800, 0x20000, 0x800, 0x40, 0x0},
 	/* Mircon MT29H8G08ACAH1 1GB */
 	{0x2c38, 0x800, 0x80000, 0x1000, 0xe0, 0x0},
-#else
-	/* Samsung K9F2G08U0M 256MB */
-	{0xecda, 0x800, 0x20000, 0x800, 0x40, 0x0},
-	/* Samsung K9F2G08U0A 256MB */
-	{0xecaa, 0x800, 0x20000, 0x800, 0x40, 0x0},
-	/* Micron MT29F2G16AAB 256MB */
-	{0x2cca, 0x800, 0x20000, 0x800, 0x40, 0x1},
-	/* Micron MT29F2G08AAC 256MB  */
-	{0x2cda, 0x800, 0x20000, 0x800, 0x40, 0x0},
-	/* Micron MT29F2G08ABD 256MB */
-	{0x2caa, 0x800, 0x20000, 0x800, 0x40, 0x0},
+#ifndef CONFIG_AT91SAM9260EK
 	/* Hynix HY27UF082G2A 256MB */
 	{0xadda, 0x800, 0x20000, 0x800, 0x40, 0x0},
 	/* Hynix HY27UF162G2A 256MB */
 	{0xadca, 0x800, 0x20000, 0x800, 0x40, 0x1},
-	/* Mircon MT29H8G08ACAH1 1GB */
-	{0x2c38, 0x800, 0x80000, 0x1000, 0xe0, 0x0},
 	/* EON EN27LN1G08 128MB */
 	{0x92f1, 0x400, 0x20000, 0x800, 0x40, 0x0},
 #endif
-- 
1.9.1

